c/meterpreter/source/metsrv/server_pivot_named_pipe.c

```diff
@@ -163,18 +163,18 @@ static DWORD read_pipe_to_packet(NamedPipeContext* ctx, LPBYTE source, DWORD sou
 					dprintf("[PIPE] Request ID found and matches expected value");
 					// we have a response to our session guid request
 					DWORD sessionGuidLen = 0;
-					LPBYTE sessionGuid = packet_get_tlv_value_raw(packet, TLV_TYPE_SESSION_GUID, 0);
+					LPBYTE sessionGuid = packet_get_tlv_value_raw(packet, TLV_TYPE_SESSION_GUID, &sessionGuidLen);
+					if (sessionGuid != NULL && sessionGuidLen == 16 && memcmp(&ctx->pivot_session_guid, sessionGuid, sizeof(ctx->pivot_session_guid)) != 0)
```

Comment:
Instead of the magic value of 16 here, this should probably use the size of the GUID in `ctx->pivot_session_guid` like:

```suggestion
					if (sessionGuid != NULL && sessionGuidLen == sizeof(ctx->pivot_session_guid) && memcmp(&ctx->pivot_session_guid, sessionGuid, sizeof(ctx->pivot_session_guid)) != 0)
```